extensions/eda/plugins/event_source/kafka.py

```diff
@@ -119,8 +122,24 @@ async def main(  # pylint: disable=R0914
             except UnicodeError:
                 logger.exception("Unicode Error")
 
+            if with_headers:
+                headers = None
+                try:
+                    headers_dict = {
+                        header[0]: header[1].decode(encoding) for header in msg.headers
+                    }
+                    headers_json = json.dumps(headers_dict)
+                    headers = json.loads(headers_json)
```

Comment:
Why do you serialize and deserialize the data? It looks like an unnecessary use of cycles per event.  
Since you are composing a python dict, we could send it directly into event["headers"]

Reply:
I am new for python. Yes. do not need to serialize and deserialize the data. Updated.